enhancements/network/core-network-policies.md

```diff
@@ -0,0 +1,547 @@
+
+3. If it starts with `openshift-` it will be treated as an OpenShift
+   namespace (since that namespace prefix has special meaning already
+   elsewhere in the code)
+
+  1. The namespace object will be changed to have a label applied with
+     the name `security.openshift.io/openshift-namespace` and the value ''
+
+4.  If it does not start with `openshift-` then any
+    `security.openshift.io/openshift-namespace` label will be stripped out and can not be set
+
+
+#### Applying the label at upgrade
+
+1. When an upgrade happens, a namespace may not be changed, so the
+   Admission Controller would not be able to apply the label
+
+2. When the Cluster Network Operator starts it would:
+
+   1. Loop over all namespaces
+
+   2. If it sees a namespace starting with `openshift-`
+
+       1. The namespace object will be changed to have a label applied
+          with the name `security.openshift.io/openshift-namespace` and the value ""
+       
+   3. Otherwise it will strip the `security.openshift.io/openshift-namespace` label
+
+   TODO: Decide if this is the correct behavior... do we want to allow
+   namespaces to opt-in to being part of the platform.  I do not think
+   there is a security issue since we are just adding restrictions,
+   but that needs more consideration.  The advantage would be for
+   things like ACS that do not install into an `openshift-` namespace.
+
+
+#### Opting Into a Default Deny for OpenShift
+
+1. The Cluster Administrator can deploy an Admin Network Policy we
+   will document and test
+
+2. The policy will change the default for all namespaces with the
+   label described above so that ingress and egress traffic will fail if
+   there is no explicit Network Policy defined for a pod in those
+   namespaces
+
+#### Setting a Global Default Deny
+
+1. The Cluster Administrator can deploy an AdminNetworkPolicy we
+   will document and test
+
+2. The policy will change the default for all namespaces in the
+   cluster so that ingress and egress traffic will fail if there is no
+   explicit Network Policy defined for a pod in those namespaces
+
+
+#### Removing Policy Restrictions for OpenShift
+
+1. The Cluster Administrator can deploy an AdminNetworkPolicy we
+   will document with a public KCS and test
+
+2. The policy will bypass Network Policy for all namespaces with the
+   label described above so that ingress and egress traffic will succeed
+   without restriction (this is the current behavior)
+
+3. This is only intended to be an escape hatch that support might need
+   to deploy until an appropriate targeted policy can be applied
+
+
+#### Developing and Shipping Network Policy
+
+1. OpenShift developers and Third-Party developers will both need to
+   create network policies for the pods running in their namespaces
+
+2. Developing policies:
+
+    1. Document what traffic flows (ingress and egress) need to be
+       allowed for pods in the namespaces
```

Comment:
How should this look for traffic flows that depend on user configuration? I.e. Cluster Monitoring Operator offers to configure Prometheus remote write (i.e. egrees traffic). Is CMO expected to dynamically create the respective NP or should this be left to the user?
Perhaps this scenario should be explicitly mention in the enhancement.